src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,171 @@
+import _ from 'underscore';
+import React from 'react';
+import {View, TouchableOpacity, Pressable} from 'react-native';
+import moment from 'moment';
+import Text from '../Text';
+import ArrowIcon from './ArrowIcon';
+import styles from '../../styles/styles';
+import {propTypes as calendarPickerPropType, defaultProps as defaultCalendarPickerPropType} from './calendarPickerPropTypes';
+import generateMonthMatrix from './generateMonthMatrix';
+import withLocalize from '../withLocalize';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import CONST from '../../CONST';
+import getButtonState from '../../libs/getButtonState';
+import * as StyleUtils from '../../styles/StyleUtils';
+
+class CalendarPicker extends React.PureComponent {
```

Comment:
I agree. It would have been good to get some wider input before introducing large changes like this.

Reply:
It was separated to two components as initially we had a logic which was hiding and opening the date picker once user focused into the date input. 

Down the road we have decided to remove this logic and the calendar is permanently shown now so it should be easier to keep it all in one component now

Reply:
Made an issue to track the improvements [here](https://github.com/Expensify/App/issues/20630) if you want to follow along

There was a bunch of follow up PRs to this component since this PR got merged so I am sure we could use some clean up and refactoring as well.